### PR TITLE
TW-130: add image pickers with multiple selections

### PR DESCRIPTION
**Pull request by sherlockvn**
https://github.com/linagora/twake-on-matrix/assets/43041967/50c71d38-d474-436f-903d-987fbb75634a

https://github.com/linagora/twake-on-matrix/assets/43041967/68f8ddbc-9aa1-4fc5-8a7d-a5c7d7e6dfaa

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -25,6 +24,14 @@
     <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW" />
     <uses-permission android:name="android.permission.READ_CONTACTS" />  
     <uses-permission android:name="android.permission.WRITE_CONTACTS" />  
+        <uses-permission android:name="android.permission.INTERNET"/>
```
Comment (imGok) [inline android/app/src/main/AndroidManifest.xml:27]: Fix indentation please :D

Comment (hoangdat) [reply android/app/src/main/AndroidManifest.xml:27]: no, it is belong to android project, not the same with flutter

**Review by imGok — APPROVED** [review]

File: android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -25,6 +24,14 @@
...
+    <uses-permission
+        android:name="android.permission.WRITE_EXTERNAL_STORAGE"
+        android:maxSdkVersion="29"/>
```
Comment (hoangdat) [inline android/app/src/main/AndroidManifest.xml:30]: IMO, we dont need write storage ATM.

File: lib/config/themes.dart [hunk 3]
```diff
@@ -267,6 +267,17 @@ abstract class FluffyThemes {
         ),
         navigationRailTheme: NavigationRailThemeData(
           indicatorColor: LinagoraSysColors.material().secondaryContainer,
+        ),
+        bottomSheetTheme: BottomSheetThemeData(
+          backgroundColor: brightness == Brightness.light 
+            ? LinagoraSysColors.material().background
+            : LinagoraSysColors.material().backgroundDark,
+          modalBackgroundColor: brightness == Brightness.light
```
Comment (hoangdat) [inline lib/config/themes.dart:275]: what is `modal`?

File: lib/pages/chat/chat.dart [hunk 4]
```diff
@@ -224,11 +229,18 @@ class ChatController extends State<Chat> {
     }
   }
 
+  void registerListenerForSelectedImagesChanged() {
```
Comment (hoangdat) [inline lib/pages/chat/chat.dart:232]:
```suggestion
  void _registerListenerForSelectedImagesChanged() {
```

File: lib/pages/chat/chat_input_row.dart [hunk 5]
```diff
@@ -350,3 +258,109 @@ class _ChatAccountPicker extends StatelessWidget {
...
+              style: Theme.of(context).textTheme.titleMedium,),
+            const Icon(Icons.chevron_right),
+            const Expanded(child: SizedBox.shrink()),
+            const Icon(Icons.more_vert),
+          ],
+        ),
+      );
+    },
+    noImagesWidget: const Center(
+      child: Text("No images found"),
```
Comment (hoangdat) [inline lib/pages/chat/chat_input_row.dart:288]: dont forget to do l10n

**Comment by nqhhdev** [comment]
Pls demo case Permission `isPermanentlyDenied`, What will happen?

**Comment by nqhhdev** [comment]
IMO: we should care about `limited`. Compare it with other apps to see how it behaves

WDYT: @hoangdat @sherlockvn

Screenshot 2023-06-14 at 00 24 38

```
  ///User has authorized this application for limited access.
  /// *Only supported on iOS (iOS14+).*
  limited,
```

File: lib/config/themes.dart (hunk 3 above)
Comment (sherlockvn) [reply lib/config/themes.dart:275]: no need this, only surfaceTint Color, let me remove

File: pubspec.yaml [hunk 6]
```diff
@@ -96,7 +96,7 @@ dependencies:
   linagora_design_flutter:
     git:
       url: git@github.com:linagora/linagora-design-flutter.git
-      ref: master
+      ref: add_image_picker
```
Comment (hoangdat) [inline pubspec.yaml:99]: can we go with master?

File: lib/pages/chat/chat.dart [hunk 7]
```diff
@@ -381,27 +397,30 @@ class ChatController extends State<Chat> {
...
-              ).detectFileType,
-            )
-            .toList(),
-        room: room!,
-      ),
-    );
+  Future<void> sendImages() async {
+    final selectedAssets = imagePickerController.sortedSelectedAssets;
+    for (final entity in selectedAssets) {
+      await sendImage(entity);
```
Comment (hoangdat) [inline lib/pages/chat/chat.dart:419]: ok for now, but we should do it better in the ticket of sending

**Review by hoangdat — APPROVED** [review]

**Pull request merged**